src/components/coin-row/FastTransactionCoinRow.tsx

```diff
@@ -0,0 +1,273 @@
+        ? contactAddress
+        : abbreviations.address(contactAddress, 4, 10);
+      contactColor = getRandomColor();
+    }
+
+    const blockExplorerAction = lang.t('exchange.coin_row.view_on', {
+      blockExplorerName: startCase(ethereumUtils.getBlockExplorer(network)),
+    });
+    if (hash) {
+      let buttons = [
+        ...(canBeResubmitted ? [TransactionActions.speedUp] : []),
+        ...(canBeCancelled ? [TransactionActions.cancel] : []),
+        blockExplorerAction,
+        ...(ios ? [TransactionActions.close] : []),
+      ];
+      if (showContactInfo) {
+        buttons.unshift(
+          contact
+            ? TransactionActions.viewContact
+            : TransactionActions.addToContacts
+        );
+      }
+
+      showActionSheetWithOptions(
+        {
+          cancelButtonIndex: buttons.length - 1,
+          options: buttons,
+          title: pending
+            ? `${headerInfo.type}${
+                showContactInfo
+                  ? ' ' + headerInfo.divider + ' ' + headerInfo.address
+                  : ''
+              }`
+            : showContactInfo
+            ? `${headerInfo.type} ${date} ${headerInfo.divider} ${headerInfo.address}`
+            : `${headerInfo.type} ${date}`,
+        },
+        (buttonIndex: number) => {
+          const action = buttons[buttonIndex];
+          switch (action) {
+            case TransactionActions.viewContact:
+            case TransactionActions.addToContacts:
+              navigate(Routes.MODAL_SCREEN, {
+                address: contactAddress,
+                asset: item,
+                color: contactColor,
+                contact,
+                type: 'contact_profile',
+              });
+              break;
+            case TransactionActions.speedUp:
+              navigate(Routes.SPEED_UP_AND_CANCEL_SHEET, {
+                tx: item,
+                type: 'speed_up',
+              });
+              break;
+            case TransactionActions.cancel:
+              navigate(Routes.SPEED_UP_AND_CANCEL_SHEET, {
+                tx: item,
+                type: 'cancel',
+              });
+              break;
+            case TransactionActions.close:
+              return;
+            case blockExplorerAction:
+              ethereumUtils.openTransactionInBlockExplorer(hash, network);
+              break;
+            default: {
+              return;
+            }
+          }
+        }
+      );
+    }
+  }, [accountAddress, contact, item, navigate]);
+
+  return (
+    <ButtonPressAnimation onPress={onPressTransaction} scaleTo={0.96}>
+      <RenderProfiler name="CoinRow" update>
+        <View style={cx.wholeRow}>
+          <FastCoinIcon
+            address={mainnetAddress || item.address}
+            assetType={item.assetType}
+            symbol={item.symbol}
+            theme={theme}
+          />
+          <View style={cx.column}>
+            <TopRow {...item} theme={theme} />
```

Comment:
never do `{...item}` unless you know for sure it's only gonna pass props you need
otherwise it's gonna just pass lots of unnecessary stuff as props and the component gonna rerender because of the change of some random value of the `item` even thought that component don't even use it